server/src/main/java/org/apache/iotdb/db/mpp/plan/analyze/AnalyzeVisitor.java

```diff
@@ -437,6 +443,69 @@ private void analyzeLastSource(
     analysis.setSourceExpressions(sourceExpressions);
   }
 
+  private ISchemaTree findAllViewsInTreeThenReFetchAndMerge(ISchemaTree originSchemaTree) {
+    PathPatternTree patternTree = new PathPatternTree();
+    GetSourcePathsVisitor getSourcePathsVisitor = new GetSourcePathsVisitor();
+    boolean needToReFetch = false;
+    try {
+      Pair<List<MeasurementPath>, Integer> tempPair =
+          originSchemaTree.searchMeasurementPaths(new PartialPath("root.**"));
+      for (MeasurementPath measurementPath : tempPair.left) {
+        if (measurementPath.getMeasurementSchema().isLogicalView()) {
+          LogicalViewSchema logicalViewSchema =
+              (LogicalViewSchema) measurementPath.getMeasurementSchema();
+          ViewExpression viewExpression = logicalViewSchema.getExpression();
+          List<PartialPath> pathsNeedToReFetch =
+              getSourcePathsVisitor.process(viewExpression, null);
+          for (PartialPath path : pathsNeedToReFetch) {
+            patternTree.appendFullPath(path);
+            needToReFetch = true;
+          }
+        }
+      }
+    } catch (Exception e) {
+      throw new RuntimeException(e);
+    }
+    if (needToReFetch) {
+      ISchemaTree viewSchemaTree = this.schemaFetcher.fetchSchema(patternTree, null);
+      originSchemaTree.mergeSchemaTree(viewSchemaTree);
+    }
+    return originSchemaTree;
```

Comment:
It is time-consuming to construct measurementPath. We can implement this check via a new method SchemaTree.hasLogicalView which can be get from the schema tree construction process.

Reply:
Added a judgment, skip this step if it does not contain a view